java-strings/src/main/java/com/baeldung/string/SubstringPalindrome.java

```diff
@@ -0,0 +1,90 @@
+package com.baeldung.string;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SubstringPalindrome {
+
+    public Set<String> isPalindromeUsingCenter(String input) {
+        final Set<String> palindromes = new HashSet<>();
+        if (input == null || input.isEmpty()) {
+            return palindromes;
+        }
+        if (input.length() == 1) {
+            palindromes.add(input);
+            return palindromes;
+        }
+        for (int i = 0; i < input.length(); i++) {
+            palindromes.addAll(findPalindromes(input, i, i + 1));
+            palindromes.addAll(findPalindromes(input, i, i));
+        }
+        return palindromes;
+    }
+
+    private Set<String> findPalindromes(String input, int low, int high) {
+        Set<String> result = new HashSet<>();
+        while (low >= 0 && high < input.length() && input.charAt(low) == input.charAt(high)) {
+            result.add(input.substring(low, high + 1));
+            low--;
+            high++;
+        }
+        return result;
+    }
+
+    public Set<String> isPalindromeUsingSubstring(String input) {
```

Comment:
A better method name would be _findAllPalindromesUsingBruteForceApproach_

Reply:
Changed name